pkg/daemon/update.go

```diff
@@ -400,6 +403,11 @@ func getFileOwnership(file ignv2_2types.File) (int, int, error) {
 	return uid, gid, nil
 }
 
+func (dn *Daemon) updateOS(oldConfig, newConfig *mcfgv1.MachineConfig) error {
+	glog.Infof("Updating OS to %s", newConfig.Spec.OSImageURL)
+	return runPivot(newConfig.Spec.OSImageURL)
```

Comment:
is pivot idempotent? this is going to get run times when there is not an actual os update, because everything gets refreshed if anything is different (eg if we add a file, we will run pivot with our currently booted os image url even though it's not different)

Reply:
Yes, it's idempotent (see https://github.com/ashcrow/pivot/pull/14), but I should add a check here as well so we don't waste time running it.